pkg/repstream/streampb/empty.go

```diff
@@ -154,62 +175,86 @@ func (d *DebugLogicalConsumerStatus) GetStats() DebugLogicalConsumerStats {
 	defer d.mu.Unlock()
 	return d.mu.stats
 }
+func (d *DebugLogicalConsumerStatus) RecordRecvStart() {
+	d.mu.Lock()
+	d.mu.stats.Recv.CurReceiveStart = timeutil.Now() // nolint:deferunlockcheck
+	d.mu.stats.CurrentState = ReceiveState
+	if !d.mu.stats.Ingest.CurIngestStart.IsZero() { // nolint:deferunlockcheck
+		d.mu.stats.Ingest.TotalIngestNanos += timeutil.Since(d.mu.stats.Ingest.CurIngestStart).Nanoseconds() // nolint:deferunlockcheck
+	} // nolint:deferunlockcheck
+	d.mu.stats.Recv.LastWaitNanos = 0
+	d.mu.stats.Ingest.CurIngestStart = time.Time{}
+	d.mu.Unlock() // nolint:deferunlockcheck
+}
 
-func (d *DebugLogicalConsumerStatus) RecordRecv(wait time.Duration) {
-	nanos := wait.Nanoseconds()
+func (d *DebugLogicalConsumerStatus) RecordRecv() {
 	d.mu.Lock()
+	nanos := timeutil.Since(d.mu.stats.Recv.CurReceiveStart).Nanoseconds() // nolint:deferunlockcheck
+	d.mu.stats.CurrentState = ProcessingState
 	d.mu.stats.Recv.LastWaitNanos = nanos
 	d.mu.stats.Recv.TotalWaitNanos += nanos
-	d.mu.Unlock()
+
+	d.mu.stats.Ingest.CurIngestStart = timeutil.Now() // nolint:deferunlockcheck
+	d.mu.stats.Recv.CurReceiveStart = time.Time{}
+	d.mu.Unlock() // nolint:deferunlockcheck
 }
 
 func (d *DebugLogicalConsumerStatus) SetInjectedFailurePercent(percent uint32) {
 	d.mu.Lock()
 	d.mu.injectFailurePercent = percent
-	d.mu.Unlock()
+	d.mu.Unlock() // nolint:deferunlockcheck
 }
 
 func (d *DebugLogicalConsumerStatus) RecordFlushStart(start time.Time, keyCount int64) uint32 {
-	micros := start.UnixMicro()
 	d.mu.Lock()
-	d.mu.stats.Flushes.Current.TotalKVs = keyCount
-	d.mu.stats.Flushes.Current.StartedUnixMicros = micros
+	// Reset the incremental flush stats.
+	d.mu.stats.Flushes.Last.SlowestBatchNanos = 0
+	d.mu.stats.Flushes.Last.Batches = 0
+	d.mu.stats.Flushes.Last.ProcessedKVs = 0
+	d.mu.stats.Flushes.Last.LastFlushNanos = 0
+
+	d.mu.stats.CurrentState = FlushingState
+	d.mu.stats.Flushes.Last.CurFlushStart = start
+	d.mu.stats.Flushes.Last.TotalKVs = keyCount
 	failPercent := d.mu.injectFailurePercent
-	d.mu.Unlock()
+	d.mu.Unlock() // nolint:deferunlockcheck
```

Comment:
I'd just defer it at this point; too many lines away to be relying on inspection of every line now.

Reply:
Done. I added a defer some other other chunky functions as well.